storage/innobase/row/row0sel.cc

```diff
@@ -2987,8 +2987,9 @@ bool row_sel_store_mysql_rec(byte *mysql_rec, row_prebuilt_t *prebuilt,
     mem_heap_empty(blob_heap);
   }
 
-  if (UNIV_LIKELY_NULL(prebuilt->compress_heap))
-    row_mysql_prebuilt_free_compress_heap(prebuilt);
+  if (prebuilt->compress_heap != nullptr) {
+    mem_heap_empty(prebuilt->compress_heap);
```

Comment:
If we are reusing, even emptying is a problem isn't it? I think partitions are supposed to their partition-specific blob heaps? Can you please check again?

Reply:
Sorry used the wrong button.. "not approved yet" :)

Reply:
@nitendra-bhosle can you please reply to these questions?

I don't understand the issue. Specifically why free is a problem. and empty is not?
AFAIU, empty doesn't the remove the first block. So may be the compressed heap is staying within the first block of heap.

Can you please create a blob text such that it occupies more than the first block in heap?